src/app.ts

```diff
@@ -26,60 +25,64 @@ const jsonKeyring = readFileSync(KEYRING_PATH).toString();
 const KEYRING = GearKeyring.fromJson(jsonKeyring, KEYRING_PASSPHRASE);
 
 const createVoucher = async (accountUser: HexString) => {
+  const provider = new WsProvider(NODE_ADDRESS);
   const api = await GearApi.create({
-    providerAddress: NODE_ADDRESS,
+    provider
   });
 
   const programId = GAME_ADDRESS as HexString;
   const account = decodeAddress(accountUser);
 
-  // Specify the number of issues
-  const tx = api.voucher.issue(
-    account,
-    programId,
-    15 * 10 ** api.registry.chainDecimals[0]
-  );
+  const programs = [programId];
+  const validForOneHour = (60 * 60) / 3; // number of blocks in one hour
+  const amountForVoucher = 15 * 10 ** api.registry.chainDecimals[0];
+
+  const voucherExists = await api.voucher.exists(account, programId)
+
+  if (voucherExists) {
+    const getAccount = await api.voucher.getAllForAccount(account);
+    const voucherId = Object.keys(getAccount)[0];
 
-  const extrinsic = tx.extrinsic;
+    const balance = await api.balance.findOut(voucherId);
+    const formattedBalance = Number(balance) / 10 ** api.registry.chainDecimals[0]
+    const existentialDeposit = Number(api.existentialDeposit) / 10 ** api.registry.chainDecimals[0]
+
+    if (formattedBalance <= existentialDeposit) {
```

Comment:
What's the point of these transformations?

```suggestion
    if (Number(balance) <= Number(api.existentialDeposit)) {
```